sklearn/linear_model/logistic.py

```diff
@@ -1538,6 +1538,15 @@ def fit(self, X, y, sample_weight=None):
             raise ValueError("class_weight provided should be a "
                              "dict or 'balanced'")
 
+        # compute the class weights for the entire dataset y
+        if self.class_weight in ("auto", "balanced"):
```

Comment:
I'm not sure that it is a bug. But I don't have a strong opinion. We could also make the ``class_weight`` pass by making the list of ``Cs`` just one long. Hm maybe the current master behavior is the same as the ``class_weight="balanced_bootstrap"`` in random forests? If so then we should do it as you propose here.